awswrangler/s3/_read.py

```diff
@@ -90,48 +87,15 @@ def _apply_partitions(df: pd.DataFrame, dataset: bool, path: str, path_root: Opt
 
 def _extract_partitions_dtypes_from_table_details(response: Dict[str, Any]) -> Dict[str, str]:
     dtypes: Dict[str, str] = {}
-    if "PartitionKeys" in response["Table"]:
-        for par in response["Table"]["PartitionKeys"]:
-            dtypes[par["Name"]] = par["Type"]
+    for par in response["Table"].get("PartitionKeys", []):
+        dtypes[par["Name"]] = par["Type"]
     return dtypes
 
 
-def _union(dfs: List[pd.DataFrame], ignore_index: Optional[bool]) -> pd.DataFrame:
-    if ignore_index is None:
-        ignore_index = False
-        for df in dfs:
-            if hasattr(df, "_awswrangler_ignore_index"):
-                if df._awswrangler_ignore_index is True:  # pylint: disable=protected-access
-                    ignore_index = True
-                    break
+def _union(dfs: List[pd.DataFrame], ignore_index: bool) -> pd.DataFrame:
     cats: Tuple[Set[str], ...] = tuple(set(df.select_dtypes(include="category").columns) for df in dfs)
     for col in set.intersection(*cats):
         cat = union_categoricals([df[col] for df in dfs])
         for df in dfs:
             df[col] = pd.Categorical(df[col].values, categories=cat.categories)
     return pd.concat(objs=dfs, sort=False, copy=False, ignore_index=ignore_index)
-
-
-def _read_dfs_from_multiple_paths(
```

Comment:
Nice, thanks for cleaning this up! Indeed looks like with the executor we no longer use it